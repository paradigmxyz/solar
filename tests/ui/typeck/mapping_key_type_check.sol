type U is int;
enum E {
    A,
    B
}

library L{
    enum E1 {
        A,
        B
    }
    struct S1 {
        int x;
    }
}


contract C {
    mapping(uint => uint) m0;
    mapping(E => uint) m1;
    mapping(U => uint) m2;
    mapping(L.E1 => uint) m3;

    // TODO: enable typeck feature to trigger this error
    mapping(L.S1 => uint) m4;
}


