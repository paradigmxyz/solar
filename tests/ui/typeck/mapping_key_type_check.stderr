error: only elementary types, user defined value types, contract types or enums are allowed as mapping keys.
  --> ROOT/tests/ui/typeck/mapping_key_type_check.sol:LL:CC
   |
LL |     mapping(L.S1 => uint) m5;
   |             ^^^^

error: aborting due to 1 previous error

